src/Microsoft.OData.Core/Evaluation/ODataConventionalEntityMetadataBuilder.cs

```diff
@@ -393,6 +393,12 @@ internal override string GetOperationTitle(string operationName)
         /// </returns>
         internal override bool TryGetIdForSerialization(out Uri id)
         {
+            if(this.ResourceMetadataContext.TypeContext.NavigationSourceName == null)
+            {
+                id = null;
+                return false;
+            }
+
             id = this.ResourceMetadataContext.Resource.IsTransient ? null : this.GetId();

```

Comment:
Id could be 'null' at Line 402, but it always returns 'true'? Now, the new return could be 'false'? what's the difference?

Reply:
For a transient resource (like in aggregation) null *is* the right value for the id. In this case, the id isn't null, it just can't be computed. 

We have existing code that checks the boolean result of the TryGetIdForSerialization and handles it differently than true with a null value.